AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginIntegrationTests/DataStoreConsecutiveUpdatesTests.swift

```diff
@@ -0,0 +1,422 @@
+                  post.id == myPost.id else {
+                return
+            }
+
+            if mutationEvent.mutationType == GraphQLMutationType.create.rawValue {
+                XCTAssertEqual(post.title, myPost.title)
+                XCTAssertEqual(post.content, myPost.content)
+                XCTAssertEqual(post.rating, myPost.rating)
+                saveSyncExpectation.fulfill()
+                return
+            }
+
+            if mutationEvent.mutationType == GraphQLMutationType.update.rawValue {
+                // this should be removed once the bug is fixed
+                XCTAssertEqual(post.title, myPost.title)
+                XCTAssertEqual(post.content, myPost.content)
+                XCTAssertEqual(post.rating, myPost.rating)
+
+                // this is the expected behavior which is currently failing
+                // XCTAssertEqual(post.title, "MyUpdatedPost")
+                // XCTAssertEqual(post.content, "This is my updated post.")
+                // XCTAssertEqual(post.rating, 5)
+
+                updateSyncExpectation.fulfill()
+                return
+            }
+        }
+
+        guard try HubListenerTestUtilities.waitForListener(with: hubListener, timeout: 5.0) else {
+            XCTFail("Listener not registered for hub")
+            return
+        }
+
+        // save the post
+        let immediateSaveExpectation = expectation(description: "Post is saved")
+        Amplify.DataStore.save(myPost) { result in
+            switch result {
+            case .success:
+                immediateSaveExpectation.fulfill()
+            case .failure(let error):
+                XCTFail("Error: \(error)")
+            }
+        }
+        wait(for: [immediateSaveExpectation], timeout: networkTimeout)
+
+        // update the created post
+        let immediateUpdateExpectation = expectation(description: "Post is updated")
+        Amplify.DataStore.save(updatedPost) { result in
+            switch result {
+            case .success:
+                immediateUpdateExpectation.fulfill()
+            case .failure(let error):
+                XCTFail("Error: \(error)")
+            }
+        }
+        wait(for: [immediateUpdateExpectation], timeout: networkTimeout)
+
+        // query the updated post immediately
+        let immediateQueryExpectation = expectation(description: "Post is updated immediately")
+        Amplify.DataStore.query(Post.self, byId: updatedPost.id) { result in
+            switch result {
+            case .success(let post):
+                XCTAssertEqual(post?.title, updatedPost.title)
+                XCTAssertEqual(post?.content, updatedPost.content)
+                XCTAssertEqual(post?.rating, updatedPost.rating)
+                immediateQueryExpectation.fulfill()
+            case .failure(let error):
+                XCTFail("Error: \(error)")
+            }
+        }
+        wait(for: [immediateQueryExpectation], timeout: networkTimeout)
```

Comment:
refactor into `guard let post = queryPost(byId: updatedPost.id) else { ... }` where you XCTFail inside the queryPost helper method, or similar method that reduces code duplication for this block of code